### PR TITLE
Update to HDF5 and PEtab v2

### LINKED ISSUE #4: PEtab YAML formatting
I identified a couple of issues with the petab files for the test cases
- [ ] all observable ids have the same name as the model entity, which is afaik not valid petab
- [ ]  we should use format_version `2.0.0` as mapping tables are not available with `1.0.0`
  - [ ] change `sbml_files` to to field `model_files` with dict `language` (sbml) and `location` (file)
  - [ ] change `mapping_tables` attribute to list `mapping_files`
  - [ ] in the mapping table, I would suggest using `petab.MODEL_ENTITY_ID` (instead of `ioId`) and `petab.PETAB_ENTITY_ID` (instead of `ioValue`). I wouldn't introduce a separate `netId` column, but rather absorb this into `petab.MODEL_ENTITY_ID` using a nomenclature such as `net1.input1`. see https://github.com/PEtab-dev/PEtab/pull/538. I am also not sure whether we should make assignments to petab variables in the mapping table (model outputs to petab variables), this should happen in the condition table (which would also be beneficial to specify things like time derivatives etc, see https://github.com/PEtab-dev/PEtab/issues/585).
  
- [ ] parameter values for neural networks are provided in a separate `parameters_nn.tsv` file, how about directly creating a proper parameters table where test values are set as nominal values? 

Code to implement a majority of the proposed changes is available here https://github.com/AMICI-dev/AMICI/blob/b201f4efc354939ac1b0763c65f2e22507d086c4/tests/sciml/testsuite.py#L177

### PR DESCRIPTION
**Pull request by sebapersson**
This is a large update to the repo. As most test files have changed I provide an overview here.

## Changes in neural network tests

The neural networks tests (net tests) are still the same. The difference is that the input, parameters and output are now provided in HDF5 file format. The saved arrays are in row-major, following PyTorch indexing. I have tested and confirmed consistent results between Julia and PyTorch.

## Changes in hybrid tests

I have now updated the yaml file to version 2. Basically, now the file looks like:

```yaml
format_version: 2
problems:
  - model_files:
      location: "lv.xml"
      language: "sbml"
    measurement_files:
      - "measurements.tsv"
    observable_files:
      - "observables.tsv"
    condition_files:
      - "conditions.tsv"
    mapping_files:
      - "mapping_table.tsv"
extensions:
  petab_sciml:
    net1:
      file: "net1.yaml"
      hybridization:
        output: "ode"
        input: "ode"
      parameters: "net1_ps.h5"
parameter_file: "parameters_ude.tsv"
```

As discussed each network now has its own parameter file, e.g. `net1_ps.h5`. Additionally, I have:

* Fixed proper PEtab formatting for the observable table
* Changed such that neural network input is hdf5 (see test case 015 and 016)
* I have updated the mapping table to follow the format in #4.

Note that settings for neural networks (such as whether to estimate the parameters or not) can be specified in the parameters table. The only thing left is the point regarding condition table in #4, but I will open a separate issue on this.

I have run all the test-cases through PEtab.jl.

@FFroehlich and @dilpath if this looks good I will merge this, close #4, and open a new issue with the points that are left to address.

### CODE REVIEW

File: test_cases/001/petab/mapping_table.tsv [hunk 1]
```diff
@@ -1,4 +1,4 @@
-netId	ioId	ioValue
-net1	input1	prey
-net1	input2	predator
-net1	output1	gamma
+petab.MODEL_ENTITY_ID	petab.PETAB_ENTITY_ID
```
Comment (dilpath) [inline test_cases/001/petab/mapping_table.tsv:1]: typos

**Review by dilpath — APPROVED** [review]
Looks good to me! Are there read/write model+HDF5 parameter methods for Python yet? If not, can assign that to me.

File: test_cases/001/petab/mapping_table.tsv (hunk 1 above)
Comment (sebapersson) [reply test_cases/001/petab/mapping_table.tsv:1]: Maybe I need new glasses, but what is the typo?

**Comment by sebapersson** [comment]
> Looks good to me! Are there read/write model+HDF5 parameter methods for Python yet? If not, can assign that to me.

I have functions for reading parameters when testing against Julia, basically in the HDF5 the parameters are already in PyTorch order, but I can to the TODO list to add more formal functions for this (see #11 and testing the roundtrip)

File: test_cases/001/petab/mapping_table.tsv (hunk 1 above)
Comment (sebapersson) [reply test_cases/001/petab/mapping_table.tsv:1]: Ohh following #4 I thought this was the syntax..., but thanks for noticing I will change!

Comment (sebapersson) [reply test_cases/001/petab/mapping_table.tsv]: Next time I should read the spec more carefully, but now fixed :)

**Pull request merged**